src/DripsDeployer.sol

```diff
@@ -319,94 +315,51 @@ contract RepoDriverModule is CallerDependentModule, DriverModule(3) {
+
+    /// @notice Deploys a contract using CREATE3.
     /// @param amount The amount to pass into the deployed contract's constructor.
-    /// @param salt The salt to use. It must have never been used by this contract.
-    /// @param initCode The init code of the deployed contract.
-    function deploy(uint256 amount, bytes32 salt, bytes memory initCode) internal {
-        (address proxy, address addr) = _computeAddress(salt, address(this));
-        require(!proxy.isContract(), "Salt already used");
-        require(amount <= address(this).balance, "Balance too low");
+    /// @param salt The deployer-specific salt for determining the deployed contract's address.
+    /// @param creationCode The creation code of the contract to deploy.
+    function deploy(uint256 amount, bytes32 salt, bytes memory creationCode) internal {
         // slither-disable-next-line unused-return
-        Create2.deploy(0, salt, PROXY_CREATION_CODE);
-        bool success;
-        // slither-disable-next-line low-level-calls
-        (success,) = proxy.call{value: amount}(initCode);
-        require(addr.isContract(), "Deployment failed");
-    }
-
-    /// @notice Computes the deterministic address of a contract deployed by this contract.
-    /// The deployed contract doesn't need to be deployed yet, it's a hypothetical address.
-    /// @param salt The salt used for deployment.
-    /// @return addr The deployed contract's address.
-    function computeAddress(bytes32 salt) internal view returns (address addr) {
-        return computeAddress(salt, address(this));
-    }
-
-    /// @notice Computes the deterministic address of a contract deployed by a deployer.
-    /// The contract doesn't need to be deployed yet, it's a hypothetical address.
-    /// @param salt The salt used for deployment.
-    /// @param deployer The address of the deployer of the proxy and the contract.
-    /// @return addr The deployed contract's address.
-    function computeAddress(bytes32 salt, address deployer) internal pure returns (address addr) {
-        (, addr) = _computeAddress(salt, deployer);
-    }
-
-    /// @notice Computes the deterministic address of a proxy and a contract deployed by a deployer.
-    /// The proxy and the contract don't need to be deployed yet, these are hypothetical addresses.
-    /// @param salt The salt used for deployment.
-    /// @param deployer The address of the deployer of the proxy and the contract.
-    /// @return proxy The proxy's address.
-    /// @return addr The deployed contract's address.
-    function _computeAddress(bytes32 salt, address deployer)
-        private
-        pure
-        returns (address proxy, address addr)
-    {
-        proxy = Create2.computeAddress(salt, keccak256(PROXY_CREATION_CODE), deployer);
-        addr = address(uint160(uint256(keccak256(abi.encodePacked(hex"d694", proxy, hex"01")))));
+        _CREATE3_FACTORY.deploy{value: amount}(salt, creationCode);
```

Comment:
One question with this pattern: 
Could an attacker not use the CREATE3_FACTORY and deploy another contract with the same `salt` before Drips is deployed?

Reply:
How is that prevented?

Reply:
The attacker can do that, but the contract would be deployed under a different address: https://github.com/ZeframLou/create3-factory/blob/06ec0ff36d41853dcd4399fbe2127aef801c4077/src/CREATE3Factory.sol#L21

Reply:
Nice, this makes sense.